File: library/median_num/median_num.go
package median_num

import (
	"sort"
)

func GetMedianNum(list1, list2 []int) float64 {
	list1 = append(list1, list2...)
	sort.Ints(list1)
	length := len(list1)
	if length == 0 {
		return 0
	} else if length == 1 {
		return float64(list1[0]) / 2
	} else if length%2 == 0 {
		return float64(list1[length/2]+list1[length/2-1]) / 2
	} else {
		return float64(list1[length/2])
	}
}

File: library/median_num/median_num.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Time    : 2020/3/22 下午1:23
# @Author  : serendipity-xp
# @FileName: median_num.py
# @Software: PyCharm
# @GitHub  ：https://github.com/serendipity-xp

class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        sum_list = list(nums1 + nums2)
        sum_list.sort()
        length = len(sum_list)
        result = float(sum_list[length / 2])
        if length % 2 == 0:
            result = float(sum_list[length / 2] + sum_list[length / 2 - 1]) / 2
        return result


if __name__ == '__main__':
    list1 = [3, 1, 5]
    list2 = [2]
    solution_obj = Solution()
    print solution_obj.findMedianSortedArrays(list1, list2)

